server/src/main/java/org/opensearch/rest/action/search/RestSearchAction.java

```diff
@@ -235,13 +235,14 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
             searchSourceBuilder.query(queryBuilder);
         }
 
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        String from = request.param("from");
+        if (from != null) {
+            searchSourceBuilder.from(Integer.parseInt(from));
```

Comment:
Will this throw a NumberFormatException if `from` is not an integer value?

`paramAsInt` handles the case where the input is not an integer.

Reply:
The difference is that `paramAsInt` throws `IllegalArgumentException` rather than `NumberFormatException`, I've changed the code to revert this change, please help to take a look, thanks!